test/e2e/batch/batch_test.go

```diff
@@ -0,0 +1,194 @@
+	assertRestOk(countersURL("batch-counter", hostAddr), client)
+}
+
+func TestBatchConfigMap(t *testing.T) {
+	name := strcase.ToKebab(t.Name())
+	infinispan := createCluster(name)
+	defer testKube.DeleteInfinispan(infinispan, tutils.SinglePodTimeout)
+
+	batchScript := batchString(`create cache --template=org.infinispan.DIST_SYNC batch-cache
+	create counter --concurrency-level=1 --initial-value=5 --storage=VOLATILE --type=weak batch-counter`)
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: infinispan.Namespace,
+		},
+		Data: map[string]string{
+			batchCtrl.BatchFilename: batchScript,
+		},
+	}
+
+	if err := testKube.Kubernetes.Client.Create(ctx, configMap); err != nil {
+		panic(fmt.Errorf("Unable to create ConfigMap: %w", err))
+	}
+
+	batch := createBatch(name, infinispan.Name, "", name)
+	defer testKube.DeleteBatch(batch)
+
+	waitForValidBatchPhase(name, v2.BatchSucceeded)
+	waitForK8sResourceCleanup(name)
+
+	client := httpClient(infinispan)
+	hostAddr := hostAddr(client, infinispan)
+	assertRestOk(cachesURL("batch-cache", hostAddr), client)
+	assertRestOk(countersURL("batch-counter", hostAddr), client)
+}
+
+func TestBatchFail(t *testing.T) {
+	name := strcase.ToKebab(t.Name())
+	infinispan := createCluster(name)
+	defer testKube.DeleteInfinispan(infinispan, tutils.SinglePodTimeout)
+
+	batchScript := "SOME INVALID BATCH CMD!"
+	batch := createBatch(name, infinispan.Name, batchScript, "")
+	defer testKube.DeleteBatch(batch)
+
+	waitForValidBatchPhase(name, v2.BatchFailed)
+	waitForK8sResourceCleanup(name)
+}
+
+func batchString(config string) string {
+	return strings.ReplaceAll(config, "\t", "")
+}
+
+func httpClient(infinispan *v1.Infinispan) tutils.HTTPClient {
+	user := consts.DefaultDeveloperUser
+	password, err := users.PasswordFromSecret(user, infinispan.GetSecretName(), tutils.Namespace, testKube.Kubernetes)
+	tutils.ExpectNoError(err)
+	protocol := testKube.GetSchemaForRest(infinispan)
+	return tutils.NewHTTPClient(user, password, protocol)
+}
+
+func cachesURL(cacheName, hostAddr string) string {
+	return fmt.Sprintf("%v/rest/v2/caches/%s", hostAddr, cacheName)
+}
+
+func countersURL(counterName, hostAddr string) string {
+	return fmt.Sprintf("%v/rest/v2/counters/%s", hostAddr, counterName)
+}
+
+func hostAddr(client tutils.HTTPClient, infinispan *v1.Infinispan) string {
+	routeName := fmt.Sprintf("%s-external", infinispan.Name)
+	return testKube.WaitForExternalService(routeName, tutils.Namespace, infinispan.GetExposeType(), tutils.RouteTimeout, client)
+}
+
+func createCluster(name string) *v1.Infinispan {
+	infinispan := tutils.DefaultSpec(testKube)
+	infinispan.ObjectMeta.Name = name
+	testKube.Create(infinispan)
+	testKube.WaitForInfinispanPods(1, tutils.SinglePodTimeout, infinispan.Name, tutils.Namespace)
+	return infinispan
+}
+
+func createBatch(name, cluster, config, configMap string) *v2.Batch {
+	stringToPtr := func(s string) *string {
```

Comment:
`pointer.StringPtr()` already implemented

Reply:
The `stringToPtr` function is different. If `s == ""`, then `nil` is returned, whereas `pointer.StringPtr()` returns a pointer to `""`.

But regardless I have just updated the function to `createBatch(name, cluster string, config, configMap *string) *v2.Batch` so neither is required.